db/blip_handler.go

```diff
@@ -320,7 +320,7 @@ func (bh *blipHandler) sendChanges(sender *blip.Sender, opts *sendChangesOptions
 				for _, item := range change.Changes {
 					changeRow := bh.buildChangesRow(change, item["rev"])
 
-					if bh.purgeOnRemoval && len(change.Removed) > 0 && bh.blipContext.ActiveProtocol() == BlipCBMobileReplicationV3 {
+					if opts.revocations && change.allRemoved && bh.blipContext.ActiveProtocol() == BlipCBMobileReplicationV3 {
```

Comment:
This looks like we're checking opts.revocations in the context of deciding whether or not to send a removal when the user still has access to it.  This seems like an overloading of the meaning of opts.revocations, which I thought was "does the peer wish to receive revocation messages for documents in a channel when they lose access to that channel".

Reply:
Well `opts.revocations` means that revocations have been requested via `subChanges`. 
If the issue is the difference between revocations and removals then this option is being used to say 'I want revocations and removals'. This was discussed with CBL.

Reply:
Although... To be honest on second though I wonder if we want to have that check at all... Should we just always send this removed message (as we did in 2.8 I think?) but then the client can decide whether to actually purge it or not.

Reply:
As discussed with CBL. The expectation is that if revocations subChanges message is not set then they fall back to the 2.8.x behaviour. This being that we don't do the user check and just send all removal messages.